asterius/rts/rts.gc.mjs

```diff
@@ -657,38 +743,54 @@ export class GC {
     }
   }
 
+  /**
+   * Allocates a new nursery and stores its address in the appropriate
+   * field of the StgRegTable of the main capability.
+   */
   updateNursery() {
+    // Note: the 'rHpAlloc' field of the 'StgRegTable' C struct contains 
+    // the number of bytes being allocated in the heap. Here, this field is
```

Comment:
More precisely, `HpAlloc` contains the number of bytes being attempted to allocate before the heap check fails.